Make the bottom-sheet cells non interactive when no action is attached directly to them

hypest opened the pull request:
Fixes #2633 

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/31052

### To test:

Please check the "How has this been tested" section in the [GB PR](https://github.com/WordPress/gutenberg/pull/31052).

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

hypest committed:
Update GB hash

hypest committed:
Update RELEASE-NOTES.txt

antonis reviewed (approved):
Approved via https://github.com/WordPress/gutenberg/pull/31052#pullrequestreview-641388672 🎉

hypest commented:
The iOS full-test failure is a known issue and fixed already in https://github.com/WordPress/gutenberg/pull/31053. Will run again just to be sure, after merging the Gutenberg PR and updating from there. Edit: updated and set the PR to auto-merge-on-green-CI.

hypest committed:
Update the GB hash to the merged-to-trunk one

hypest committed:
Merge branch 'develop' into issue/2633-remove-press-action-on-non-interactive-cell-roots
